hack/scripts/sync-prow.sh

```diff
@@ -3,30 +3,40 @@
 set -euo pipefail
 
 check_deps() {
-  if ! which git-filter-repo > /dev/null 2>&1; then
+  if ! command -v git-filter-repo &>/dev/null; then
     echo "please install the git-filter-repo tool per https://github.com/newren/git-filter-repo"
     exit 1
   fi
+
+  if [[ -z "${1}" ]]; then
+    echo "please specify your github username as a positional parameter."
+    echo "${1}"
+    exit 1
+  fi
 }
 
-check_deps
+# $1 must be your GitHub username.
+check_deps "${1:-}"
 
-old_prow="$PWD/test-infra"
+gh_username="${1}"
+old_prow="$PWD/old_prow"
 new_prow="$PWD/prow"
 
 # Clone the old prow repo.
-if [[ ! -d "$old_prow" ]]; then
-    git clone https://github.com/kubernetes/test-infra.git
-fi
+git clone https://github.com/kubernetes/test-infra.git $old_prow || true
```

Comment:
It seems git clone could fail for a variety of reasons, some fatal (e.g. network error?) and the "true" alternation would mask such events. It seems the predicate approach is better since it only tries if not already present, which I presume is the error case being considered. (Same below.)

Reply:
If the repo fails to clone subsequent steps will fail in a way that makes that clear. This sync script only needs to be run a few times so a network error is unlikely to occur. I don't really want to spend time implementing graceful error handling or fixing more typos.
I only changed this due to the requests that were made after the PR that added this script had already merged: https://github.com/kubernetes-sigs/prow/pull/78#pullrequestreview-1846560173